internal/bminventory/inventory.go

```diff
@@ -2135,6 +2135,23 @@ func setCommonUserNetworkManagedParams(params *models.ClusterUpdateParams, singl
 	return nil, false
 }
 
+func (b *bareMetalInventory) updatePlatformSources(params installer.UpdateClusterParams, cluster *common.Cluster, updates map[string]interface{}, usages map[string]models.Usage) {
+	b.updatePlatformParams(params, cluster, updates)
+
+	if params.ClusterUpdateParams.Platform.Type != models.PlatformTypeBaremetal {
+		withCredentials := false
+
+		if params.ClusterUpdateParams.Platform.Type == models.PlatformTypeVsphere {
+			if params.ClusterUpdateParams.Platform.Vsphere != nil && params.ClusterUpdateParams.Platform.Vsphere.VCenter != nil {
+				withCredentials = true
+			}
+		}
```

Comment:
you can set it as a compound logical expression: withCredentials := params.ClusterUpdateParams.Platform.Type == models.PlatformTypeVsphere && params.ClusterUpdateParams.Platform.Vsphere != nil && params.ClusterUpdateParams.Platform.Vsphere.VCenter != nil

Reply:
It will be cleared that way